editor/plugins/sprite_frames_editor_plugin.cpp

```diff
@@ -1098,6 +1099,46 @@ void SpriteFramesEditor::_animation_add() {
 	animations->grab_focus();
 }
 
+void SpriteFramesEditor::_animation_duplicate() {
+	if (updating) {
+		return;
+	}
+
+	if (!frames->has_animation(edited_anim)) {
+		return;
+	}
+
+	int counter = 1;
+	String new_name = String(edited_anim);
+	String base_name = new_name;
+	PackedStringArray name_component = new_name.split("_");
```

Comment:
This can be simplified by using `rsplit()` with `maxsplit` being 1. It will return the last bit and the remainder of the String.